packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,540 @@
+  final Color backgroundColor;
+
+  /// Explicit list of child cell widgets. If [plainChildren] is provided then
+  /// [plainChildrenBuilder] must be null.
+  final List<Widget> plainChildren;
+
+  /// Lazily built list of child cell widgets. If [plainChildrenBuilder] is
+  /// provided then [plainChildren] must be null.
+  final IndexedWidgetBuilder plainChildrenBuilder;
+
+  /// The number of children that can be built with [plainChildrenBuilder].
+  ///
+  /// The [plainChildCount] is optional. If it's provided then only a subset
+  /// of the plain children will be built at a time, based on viewport size. If
+  /// [plainChildCount] is not provided, then all plain children will be
+  /// created from the very beginning.
+  final int plainChildCount;
+
+  @override
+  _CupertinoTableViewState createState() => _CupertinoTableViewState();
+}
+
+class _CupertinoTableViewState extends State<CupertinoTableView> {
+  // Controller that reports the size of the last cell in the table, used to
+  // determine the desired gap between all extra dividers below it.
+  final _CellHeightController _cellHeightController = _CellHeightController();
+
+  // Decorator that builds cell widgets with extra chrome and/or functionality,
+  // e.g., adding a divider to a cell.
+  IndexedWidgetBuilder _cellDecorator;
+
+  // Cache of undecorated cell widgets. The cache is cleared every time this
+  // widget runs `build()`. The purpose of this cache is to avoid repeatedly
+  // building every cell twice while searching for the last cell in the list.
+  // This cache map structure is "index of cell" => "widget for cell".
+  final Map<int, Widget> _cellWidgetCache = <int, Widget>{};
+
+  @override
+  void initState() {
+    super.initState();
+    _cellDecorator = _createCellDecorator();
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTableView oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _cellDecorator = _createCellDecorator();
+  }
+
+  @override
+  void reassemble() {
+    super.reassemble();
+    _cellDecorator = _createCellDecorator();
+  }
+
+  IndexedWidgetBuilder _createCellDecorator() {
+    if (widget.plainChildren != null) {
+      return _buildPlainChildFromList;
+    } else if (widget.plainChildrenBuilder != null) {
+      return _buildPlainChildFromBuilder;
+    } else {
+      throw Exception('No children found.');
+    }
+  }
+
+  Widget _buildPlainChildFromList(BuildContext context, int index) {
+    if (index >= widget.plainChildren.length) {
+      return null;
+    }
+
+    final bool isLastCell = index == widget.plainChildren.length - 1;
+    return _decorateCell(widget.plainChildren[index], isLastCell);
+  }
+
+  Widget _buildPlainChildFromBuilder(BuildContext context, int index) {
+    // If this cell has already been built since the last time `build()` was
+    // called, return that cached cell widget.
+    if (_cellWidgetCache[index] != null) {
+      return _cellWidgetCache[index];
```

Comment:
Do you need to decorate this? Add test if you keep this structure.

Reply:
This is a cache of the cells, themselves, not the decorated cells.  What is the behavior here that you think requires verification in a test?

Reply:
I might not be reading this right, but this function seems to intend to return a decorated cell based on `return _decorateCell(cell, isLastCell)`. `cell` there is an undecorated cell which you're caching. Though if you hit the cache the next time, you return that undecorated cell as is without decoration. Don't know if that's intentional.

Reply:
Good point. I reorganized the logic so that whether a cell is from the cache or builder, it now has the same execution path with regard to decoration.